rpc/src/request.rs

```diff
@@ -38,13 +38,30 @@ impl<R> Wrapper<R>
 where
     R: Request,
 {
-    /// Create a new request wrapper from the given request
+    /// Create a new request wrapper from the given request.
+    ///
+    /// By default this sets the ID of the request to a random [UUIDv4] value.
+    ///
+    /// [UUIDv4]: https://en.wikipedia.org/wiki/Universally_unique_identifier
     pub fn new(request: R) -> Self {
+        Wrapper::new_with_id(Id::uuid_v4(), request)
+    }
+
+    /// Create a new request wrapper with a custom JSONRPC request ID.
+    pub fn new_with_id(id: Id, request: R) -> Self {
```

Comment:
Since the ID of incoming events to the client correlates with the ID of the JSON-RPC wrapper for the initial subscription request, I initially needed this constructor to be able to control that ID.

I've since refactored this in a way that we no longer need this constructor.